src/framework/auth/providers/firebase-auth.provider.ts

```diff
@@ -0,0 +1,220 @@
+ *
+ * ```
+ * {
+ *  firebase: {
+ *    apiKey: 'your firebase api key',
+ *    authDomain: 'your firebase auth domain',
+ *    databaseURL: 'your firebase database url',
+ *    projectId: 'your firebase project id',
+ *    storageBucket: 'your firebase storage bucket',
+ *    messagingSenderId: 'your firebase messaging sender id'
+ *   },
+ *  login: {
+ *    redirect: {
+ *      success: '/',
+ *      failure: null,
+ *    }
+ *  },
+ *  register: {
+ *    redirect: {
+ *      success: '/',
+ *      failure: null,
+ *    }
+ *  },
+ *  logout: {
+ *    redirect: {
+ *      success: '/',
+ *      failure: null,
+ *    },
+ *  requestPass: {
+ *    redirect: {
+ *      success: '/',
+ *      failure: null,
+ *    },
+ *  resetPass: {
+ *    redirect: {
+ *      success: '/',
+ *      failure: null,
+ *    }
+ *  }
+ *}
+ *
+ * // Note, there is no need to copy over the whole object to change the settings you need.
+ * // Also, this.getConfigValue call won't work outside of the default config declaration
+ * // (which is inside of the `NbFirebaseAuthProvider` class), so you have to replace it with a custom helper function
+ * // if you need it.
+ * ```
+ */
+@Injectable()
+export class NbFirebaseAuthProvider extends NbAbstractAuthProvider {
+
+  protected defaultConfig: NgEmailPassAuthProviderConfig = {
+    login: {
+      redirect: {
+        success: '/',
+        failure: null,
+      },
+    },
+    register: {
+      redirect: {
+        success: '/',
+        failure: null,
+      },
+    },
+    requestPass: {
+      redirect: {
+        success: '/auth/login',
+        failure: null,
+      },
+    },
+    resetPass: {
+      redirect: {
+        success: '/auth/login',
+        failure: '/auth/reset-password',
+      },
+    },
+    logout: {
+      redirect: {
+        success: '/auth/login',
+        failure: null,
+      },
+    },
+  };
+
+
+  /**
+   *The method initializes the 'firebase' object with user parameters.
+   *
+   */
+  init() {
+    firebase.initializeApp(this.getConfigValue('firebase'));
+  }
+
+  /**
+   * Firebase authentication.
+   *
+   * @param data any
+   * @returns Observable<NbAuthResult>
+   */
+  authenticate(data?: any): Observable<NbAuthResult> {
+    return Observable.fromPromise(firebase.auth().signInWithEmailAndPassword(data.email, data.password))
+      .map((res) => {
+        return this.processSuccess(res, this.getConfigValue('login.redirect.success'), [res.message]);
+      })
+      .catch((res) => {
+        return Observable.of(
+          this.processFailure(res, this.getConfigValue('login.redirect.failure'), [res.message]),
+        );
+      });
+  }
+
+  /**
+   * Firebase registration.
+   *
+   * @param data any
+   * @returns Observable<NbAuthResult>
+   */
+  register(data?: any): Observable<NbAuthResult> {
+    return Observable.fromPromise(firebase.auth().createUserWithEmailAndPassword(data.email, data.password))
+      .map((res) => {
```

Comment:
should be flatMap? 
otherwise the result is Observable<NbAuthResult| Obsevable<NbAuthResult>>